plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,458 @@
+        assertUpdate("INSERT INTO " + tableName + "(test) VALUES 'Hello', U&'hello\\6d4B\\8Bd5\\+10FFFFworld\\7F16\\7801' ", 2);
+        assertThat(computeActual("SELECT test FROM " + tableName).getOnlyColumnAsSet())
+                .containsExactlyInAnyOrder("Hello", "hello测试􏿿world编码");
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate("CREATE TABLE " + tableName + "(test varchar, ignore varchar) with (primary_key = ARRAY['test'])");
+        assertUpdate("INSERT INTO " + tableName + "(test) VALUES 'aa', 'bé'", 2);
+        assertQuery("SELECT test FROM " + tableName, "VALUES 'aa', 'bé'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test = 'aa'", "VALUES 'aa'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test > 'ba'", "VALUES 'bé'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test < 'ba'", "VALUES 'aa'");
+        assertQueryReturnsEmptyResult("SELECT test FROM " + tableName + " WHERE test = 'ba'");
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate("CREATE TABLE " + tableName + "(test varchar, ignore varchar) with (primary_key = ARRAY['test'])");
+        assertUpdate("INSERT INTO " + tableName + "(test) VALUES 'a', 'é'", 2);
+        assertQuery("SELECT test FROM " + tableName, "VALUES 'a', 'é'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test = 'a'", "VALUES 'a'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test > 'b'", "VALUES 'é'");
+        assertQuery("SELECT test FROM " + tableName + " WHERE test < 'b'", "VALUES 'a'");
+        assertQueryReturnsEmptyResult("SELECT test FROM " + tableName + " WHERE test = 'b'");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Override
+    public void testShowCreateTable()
+    {
+        assertThat((String) computeActual("SHOW CREATE TABLE orders").getOnlyValue())
+                // If the connector reports additional column properties, the expected value needs to be adjusted in the test subclass
+                .matches("CREATE TABLE \\w+\\.\\w+\\.orders \\Q(\n" +
+                        "   orderkey bigint NOT NULL,\n" +
+                        "   custkey bigint NOT NULL,\n" +
+                        "   orderstatus varchar(1) NOT NULL,\n" +
+                        "   totalprice double NOT NULL,\n" +
+                        "   orderdate date NOT NULL,\n" +
+                        "   orderpriority varchar(15) NOT NULL,\n" +
+                        "   clerk varchar(15) NOT NULL,\n" +
+                        "   shippriority integer NOT NULL,\n" +
+                        "   comment varchar(79) NOT NULL\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   primary_key = ARRAY['orderkey']\n)");
+    }
+
+    @Override
+    public void testCaseSensitiveTopNPushdown()
```

Comment:
Is it overridden for the primary key specification? Maybe we can modify the base test class to allow using a provided impl of TestTable which will add table properties.

Reply:
We can override the creation of TestTable for add table properties, but we can not avoid the test insert dup primary key or show the columns, then the test case will still be failed. then the test case still  need to modify.

The problem seems in current test system, we do not consider the connectors that has special requirements, eg:

1. Must has primary key and the value can not dup or null
2.Not support CTAS, but support CREATE, INSERT ...SELECT.